### PR TITLE
Add Dask Contrib docs

### PR DESCRIPTION
**Pull request by jacobtomlinson**
This PR adds a documentation page describing the [Dask Contrib GitHub Organization](https://github.com/dask-contrib).

I have kinda arbitrarily made up all the criteria and stuff in here with things I consider sensible to give the community some guiding points to keep in mind. However some discussion and feedback here would be much appreciated.

For example I've written it with pretty formal language, but perhaps we want to make it more informal. Perhaps we want to add/remove criteria. Perhaps we don't want any criteria at all.

### CODE REVIEW

File: docs/source/contrib.rst [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+If you are working on a project which you think could be a great addition to the Dask Contrib org then please
+feel free to raise an issue on the `Dask Community repo <https://github.com/dask/community>`_ requesting a transfer.
+
+For a project to be accepted into Dask Contrib it must meet the following criteria:
+
+- Packages must complement or enhance Dask in some way, maybe it contains plugins, adds integration with some other library or just generally improves the Dask experience for users.
+- Packages must have at least one active maintainer.
+- Packages must be installable both from PyPI with ``pip`` and Conda Forge with ``conda``.
+- Packages must have documentation.
```
Comment (jsignell) [inline docs/source/contrib.rst:25]: This is pretty package-centric, I can imagine there being examples, benchmarks, or integration tests that live in dask-contrib. But I am fine with just crossing that bridge when/if we get to it.

File: docs/source/contrib.rst [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+Graduation to the main org
+--------------------------
+
+In some cases it may be appropriate to graduate a project to the main Dask Organization, or perhaps merge the code/functionality
+into an existing Dask project.
+
+This will be assessed on a case-by-case basis by the Dask maintenance team.
```
Comment (jsignell) [inline docs/source/contrib.rst:33]: Do you mean precisely `@dask/maintenance`? If so link, and we should keep that lteam more up-to-date. If not maybe reword slightly.

File: docs/source/contrib.rst [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+For some projects it may make sense to graduate their content into other repositories. The now defunct ``dask-lightgbm`` project was integrated into
```
Comment (jsignell) [inline docs/source/contrib.rst:35]: Just a wording suggestion - "defunct" has a negative connotation to me, but I don't feel strongly about this.

```suggestion
For some projects it may make sense to graduate their content into other repositories. The now archived ``dask-lightgbm`` project was integrated into
```

File: docs/source/contrib.rst [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+LightGBM for example.
+
+Generally for a project to make it into the main Dask Org it should meet the following criteria:
+
+- Packages much have an active community.
+- Packages must be maintained by one or more people who are employed to do so and will dedicate a minimum of 1 day per week to Dask.
```
Comment (jsignell) [inline docs/source/contrib.rst:41]: I think several of the current dask repos don't meet this. That might be fine since if a PR is a puppy, a package is like a farm. But just wanted to mention this.

File: docs/source/contrib.rst [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+- Maintainers must engage in providing support for the package on Stack Overflow.
```
Comment (jsignell) [inline docs/source/contrib.rst:42]: Or in the discussions tab in github?

**Review by jsignell — COMMENTED** [review]
I really like the idea of formalizing this process. Thanks for getting this started @jacobtomlinson!

File: docs/source/contrib.rst (hunk 4 above)
Comment (jacobtomlinson) [reply docs/source/contrib.rst:41]: > I think several of the current dask repos don't meet this.

Yeah I thought that when writing this. Perhaps some projects should be demoted 😆.

File: docs/source/contrib.rst (hunk 5 above)
Comment (jacobtomlinson) [reply docs/source/contrib.rst:42]: Ah yeah!

File: docs/source/contrib.rst (hunk 2 above)
Comment (jsignell) [reply docs/source/contrib.rst:33]: oh true true. Yeah ignore the part about the link then.

File: docs/source/contrib.rst (hunk 1 above)
Comment (stsievert) [reply docs/source/contrib.rst:25]: Would the benchmarks at https://blog.dask.org/2019/09/30/dask-hyperparam-opt#performance be appropriate? The data behind these benchmarks is at https://github.com/stsievert/dask-hyperband-comparison.

Comment (jrbourbeau) [reply docs/source/contrib.rst:25]: I think it's alright to be package-centric. Things like examples and benchmarks are useful, but can also be a slippery slope if several different groups come forward with their own benchmarks. I'd prefer to keep a focus of tools, plugins, integrations, etc. for `dask-contrib` and direct examples to `dask-examples`. Though I'm certainly interested in other's thoughts too

Comment (stsievert) [reply docs/source/contrib.rst:25]: Good point @jrbourbeau. I only have a desire to use dask-contrib with the implementations, not the experiments. That's true for the Hyperband implementation above, and it's also true for my current project (I think [adadamp] should eventually go under dask-contrib, but I do not feel the same way about [adadamp-experiments]).

[adadamp]:https://github.com/stsievert/adadamp
[adadamp-experiments]:https://github.com/stsievert/adadamp-experiments

Comment (jsignell) [reply docs/source/contrib.rst:25]: This makes total sense. I am happy with the package-centric approach :)

File: docs/source/contrib.rst [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+For a project to be accepted into Dask Contrib it must meet the following criteria:
+
+- Packages must complement or enhance Dask in some way, maybe it contains plugins, adds integration with some other library or just generally improves the Dask experience for users.
+- Packages must have at least one active maintainer.
+- Packages must be installable both from PyPI with ``pip`` and Conda Forge with ``conda``.
+- Packages must have documentation.
+
+Graduation to the main org
+--------------------------
```
Comment (jrbourbeau) [inline docs/source/contrib.rst:28]:
```suggestion
Graduation to the Dask organization
------------------------------------
```

File: docs/source/contrib.rst [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+In some cases it may be appropriate to graduate a project to the main Dask Organization, or perhaps merge the code/functionality
+into an existing Dask project.
+
+This will be assessed on a case-by-case basis by the Dask maintenance team.
+
+For some projects it may make sense to graduate their content into other repositories. The now archived ``dask-lightgbm`` project was integrated into
+LightGBM for example.
+
+Generally for a project to make it into the main Dask Org it should meet the following criteria:
```
Comment (jrbourbeau) [inline docs/source/contrib.rst:38]:
```suggestion
Generally for a project to make it into the Dask GitHub organization it should meet the following criteria:
```

File: docs/source/contrib.rst [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+- Packages much have an active community.
+- Packages must be maintained by one or more people who are employed to do so and will dedicate a minimum of 1 day per week to Dask.
```
Comment (jrbourbeau) [inline docs/source/contrib.rst:41]: I'm a little conflicted over the "who are employed to do so" part of this criteria. On the one hand I think that if project maintenance is a part of one's job it's much more likely that the maintainer will stay on top of things. On the other hand there are lots of OSS volunteers who might also dedicate 1 day per week towards project maintenance.

I'm not necessarily saying we should remove the employment aspect of this criteria, but I did want to at least raise my concern about it

File: docs/source/contrib.rst (hunk 5 above)
Comment (jrbourbeau) [reply docs/source/contrib.rst:42]: +1 to not just limiting to Stack Overflow

File: docs/source/contrib.rst [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+- Maintainers must engage in providing support for the package on Stack Overflow.
+- Packages must have documentation integrated into the larger `Dask Sphinx website <https://github.com/dask/dask-sphinx-theme>`_.
```
Comment (jrbourbeau) [inline docs/source/contrib.rst:43]:
```suggestion
- Packages must have documentation which use `Dask's sphinx theme <https://github.com/dask/dask-sphinx-theme>`_.
```

**Review by jrbourbeau — COMMENTED** [review]
Thanks for working on this @jacobtomlinson! Overall this looks good -- apologies for the delayed review

File: docs/source/contrib.rst (hunk 8 above)
Comment (jacobtomlinson) [reply docs/source/contrib.rst:42]: Yeah I appreciate what you're saying.

My goal here was to indicate that there should be some firm commitment from a maintainer, not just a casual side project.

Comment (jsignell) [reply docs/source/contrib.rst:42]: Yeah I think we can scratch "who are employed to do so" it ran a little odd to my ears as well.

Comment (jrbourbeau) [reply docs/source/contrib.rst:42]: I like @jacobtomlinson's idea of a "firm commitment" -- that seems to communicate the core message that projects in the Dask org shouldn't be treated as casual side projects

Comment (stsievert) [reply docs/source/contrib.rst:42]: Are these standards evaluated continually or only on admission? Should there be a demotion policy?

I'm thinking of libraries like [dask-tensorflow][1] that satisfied most of these requirements during development. Currently, it does not satisfy any requirements for a main Dask org package and barely satisfies requirements for a Dask-contrib package.

[1]:https://github.com/dask/dask-tensorflow
[2]:https://github.com/dask/dask-lightgbm

Comment (jrbourbeau) [reply docs/source/contrib.rst:42]: That's a great question @stsievert. As of today we've never removed a project from the Dask GitHub org -- though recently we've archived several repos that were no longer maintained (e.g. dask-tensorflow is archived). That said, I get the sense there's an appetite for being able to move some projects from the main dask org into dask-contrib so we can keep them around while (trying to) set user expectations around project stability and maintenance. 

In practice I suspect we'll want to handle this on a case-by-base basis much like how we plan to move projects from dask-contrib into dask. Perhaps a lightweight sentence or two about opening up an issue in the Dask community tracker for moving a project from dask-contrib -> dask would be sufficient.

Comment (stsievert) [reply docs/source/contrib.rst:42]: > while (trying to) set user expectations around project stability and maintenance.

That'd make a *great* documentation page. That page would help make the (experimental) integrations a lot more visible to users, and especially if it points to projects that aren't in dask-contrib but make heavy use of Dask (like dask-lightgbm).x

There should also probably be a section on the different main Dask org libraries integrate (e.g., https://ml.dask.org/pytorch.html).

File: docs/source/contrib.rst [hunk 10]
```diff
@@ -0,0 +1,44 @@
+Dask Contrib
+============
+
+In addition to the primary `Dask GitHub Organization <https://github.com/dask>`_ there is also a secondary
+`Dask Contrib GitHub Organization <https://github.com/dask-contrib/>`_ which contains community supported
+projects which integrate with or enhance the Dask experience.
+
+These projects are owned and maintained by folks outside of the core Dask maintainer group and therefore
+the quality and stability may be different to that of the main Dask packages. However these projects are
+expected to conform to the `Dask Code of Conduct <https://github.com/dask/governance/blob/main/code-of-conduct.md>`_.
```
[the remark this answers is not in the document]
Comment (jrbourbeau) [reply docs/source/contrib.rst:11]: xref https://github.com/dask-contrib/.github/pull/1

**Comment by martindurant** [comment]
Note that we should also agree on who has the ability to +1 repos for inclusion and who has write access on the org (the same set as for the dask org?). Presumably the original maintainers of the new contrib package would carry forward their permissions to the moved repo.

**Comment by jacobtomlinson** [comment]
I haven't looked at this for a while. Folks are very welcome to push commits here.

**Comment by bparbhu** [comment]
So does dask-contrib function as a way to prototype new features for dask?

**Comment by martindurant** [comment]
> So does dask-contrib function as a way to prototype new features for dask?

If you follow the conversation above, you will see that we are both talking about "incubation" (repos that may eventually move to the dask/ org) and "adjunct" (repos that add a specific thing outside of what we may consider a core concern of dask). Either of these may be experimental, but being in dask-contrib already assumes a certain level of readiness. For pure POCs/prototypes, there is no need to move them to dask-contrib.

This is *not* a place for changes that are meant to be affected on the core repos, dask/dask and dask/distributed. We use PRs for that, which might be marked as draft.

**Comment by bparbhu** [comment]
Ah I see, sorry for the confusion on my part. Thanks again for the clarification @martindurant, much appreciated.

**Comment by bparbhu** [comment]
I moved the project we were talking about here just to get started. 
https://github.com/bparbhu?tab=projects

Should this topic continue under this project or should I move the project to someplace under the dask-contrib banner?
I'm not sure what would be best.

Thanks guys.

File: docs/source/contrib.rst (hunk 8 above)
Comment (jcrist) [reply docs/source/contrib.rst:42]: I think this line is the only bit that needs to be changed (dropping the employment criteria in favor of something a bit softer). @jacobtomlinson do you have time to finish this up?

File: docs/source/index.rst [hunk 11]
```diff
@@ -190,6 +190,7 @@ messy situations in everyday problems.
    phases-of-computation.rst
    remote-data-services.rst
    gpu.rst
+   contrib.rst
```
Comment (jcrist) [inline docs/source/index.rst:193]: Perhaps rename this file, since `contrib` sounds like contribution guidelines. `dask-contrib.rst`?
